### PR TITLE
Init tai at first request

### PR DESCRIPTION
**Pull request by utle**
New attribute to control whether to initialize the TAI at server start up or at the first incoming request.

### CODE REVIEW

**Comment by utle** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.spnego_fat,com.ibm.websphere.security_fat,com.ibm.ws.collective.security_fat,com.ibm.ws.collective.security_zfat,com.ibm.ws.ejbcontainer.security.3.2_fat,com.ibm.ws.ejbcontainer.security.jacc_fat_java7,com.ibm.ws.ejbcontainer.security_fat,com.ibm.ws.ejbcontainer.security_fat.features,com.ibm.ws.security.audit_fat.config,com.ibm.ws.security.audit_fat.example,com.ibm.ws.security.audit_fat.features,com.ibm.ws.security.auth.data_fat,com.ibm.ws.security.authentication.builtin_fat,com.ibm.ws.security.authentication.tai_fat,com.ibm.ws.security.authorization.saf_zfat,com.ibm.ws.security.client_fat_java7,com.ibm.ws.security.context_fat,com.ibm.ws.security.credentials.saf_zfat,com.ibm.ws.security.csiv2_fat,com.ibm.ws.security.java2sec_fat,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.quickstart_fat,com.ibm.ws.security.registry.saf_zfat,com.ibm.ws.security.registry_fat,com.ibm.ws.security.saml.sso_fat.config,com.ibm.ws.security.saml.sso_fat.config.mapToUserRegistry,com.ibm.ws.security.saml.sso_fat.endpoint.samlmetadata,com.ibm.ws.security.saml.sso_fat.jaxrs,com.ibm.ws.security.saml.sso_fat.jaxrs.config,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.spnego_fat,com.ibm.ws.security.thread.identity_fat,com.ibm.ws.security.thread.zos.context_zfat,com.ibm.ws.security.thread.zos_zfat,com.ibm.ws.security.token.ltpa_fat,com.ibm.ws.security.utility_fat,com.ibm.ws.security.wim.scim_fat,com.ibm.ws.security.wim.scim_zfat,com.ibm.ws.security_fat,com.ibm.ws.security_fat.features,com.ibm.ws.webcontainer.security.feature_fat,com.ibm.ws.webcontainer.security.jacc.1.5_fat,com.ibm.ws.webcontainer.security_fat,com.ibm.ws.webcontainer.security_fat.features,com.ibm.ws.webcontainer.security_fat.passwordutil,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.sipcontainer_fat

**Comment by utle** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.spnego_fat,com.ibm.websphere.security_fat,com.ibm.ws.collective.security_fat,com.ibm.ws.collective.security_zfat,com.ibm.ws.ejbcontainer.security.3.2_fat,com.ibm.ws.ejbcontainer.security_fat,com.ibm.ws.ejbcontainer.security_fat.features,com.ibm.ws.security.audit_fat.config,com.ibm.ws.security.audit_fat.example,com.ibm.ws.security.audit_fat.features,com.ibm.ws.security.auth.data_fat,com.ibm.ws.security.authentication.builtin_fat,com.ibm.ws.security.authentication.tai_fat,com.ibm.ws.security.authorization.saf_zfat,com.ibm.ws.security.client_fat_java7,com.ibm.ws.security.context_fat,com.ibm.ws.security.credentials.saf_zfat,com.ibm.ws.security.csiv2_fat,com.ibm.ws.security.java2sec_fat,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.quickstart_fat,com.ibm.ws.security.registry.saf_zfat,com.ibm.ws.security.registry_fat,com.ibm.ws.security.saml.sso_fat.config,com.ibm.ws.security.saml.sso_fat.config.mapToUserRegistry,com.ibm.ws.security.saml.sso_fat.endpoint.samlmetadata,com.ibm.ws.security.saml.sso_fat.jaxrs,com.ibm.ws.security.saml.sso_fat.jaxrs.config,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.spnego_fat,com.ibm.ws.security.thread.identity_fat,com.ibm.ws.security.thread.zos.context_zfat,com.ibm.ws.security.thread.zos_zfat,com.ibm.ws.security.token.ltpa_fat,com.ibm.ws.security.utility_fat,com.ibm.ws.security.wim.scim_fat,com.ibm.ws.security.wim.scim_zfat,com.ibm.ws.security_fat,com.ibm.ws.security_fat.features,com.ibm.ws.webcontainer.security.feature_fat,com.ibm.ws.webcontainer.security.jacc.1.5_fat,com.ibm.ws.webcontainer.security_fat,com.ibm.ws.webcontainer.security_fat.features,com.ibm.ws.webcontainer.security_fat.passwordutil,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.sipcontainer_fat

File: dev/com.ibm.ws.security.authentication.tai/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -59,3 +59,6 @@ extraProperties.description=Collection of properties for the interceptor.
 disableLtpaCookie=Disable LTPA cookie
 disableLtpaCookie.desc=Do not create an LTPA cookie for the trust association interceptor (TAI). 
 
+initAtFirstRequest=Initialize the TAI at the first incoming request
+initAtFirstRequest.desc=Control whether to initialize the trust association interceptor (TAI) at the first incoming request or at the server start up. The default is initializing the TAI at the server start up.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.authentication.tai/resources/OSGI-INF/l10n/metatype.properties:63]: Change both uses of "start up" to "startup" (one word). The term is used a noun both times in this message description so is "startup."

Also, consider replacing "Control" with "Choose" or "Select" and "is initializing" with "to initialize."

**Review by helyarp — COMMENTED** [review]
Changes suggested.

**Review by helyarp — COMMENTED** [review]
Please change "whether initializing" back to "whether to initialize." 

The original comment referred to the second "initializing." It's better to say "to initialize" than "initializing."

File: dev/com.ibm.ws.security.authentication.tai/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (utle) [reply dev/com.ibm.ws.security.authentication.tai/resources/OSGI-INF/l10n/metatype.properties]: Something likes below?
Select whether to initialize the trust association interceptor (TAI) at the first incoming request or at the server startup. The default is to initialize the TAI at the server startup.

**Review by helyarp — COMMENTED** [review]
Message looks good!

**Comment by utle** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.spnego_fat,com.ibm.websphere.security_fat,com.ibm.ws.collective.security_fat,com.ibm.ws.collective.security_zfat,com.ibm.ws.ejbcontainer.security.3.2_fat,com.ibm.ws.ejbcontainer.security_fat,com.ibm.ws.ejbcontainer.security_fat.features,com.ibm.ws.security.audit_fat.config,com.ibm.ws.security.audit_fat.example,com.ibm.ws.security.audit_fat.features,com.ibm.ws.security.auth.data_fat,com.ibm.ws.security.authentication.builtin_fat,com.ibm.ws.security.authentication.tai_fat,com.ibm.ws.security.authorization.saf_zfat,com.ibm.ws.security.client_fat_java7,com.ibm.ws.security.context_fat,com.ibm.ws.security.credentials.saf_zfat,com.ibm.ws.security.csiv2_fat,com.ibm.ws.security.java2sec_fat,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.quickstart_fat,com.ibm.ws.security.registry.saf_zfat,com.ibm.ws.security.registry_fat,com.ibm.ws.security.saml.sso_fat.config,com.ibm.ws.security.saml.sso_fat.config.mapToUserRegistry,com.ibm.ws.security.saml.sso_fat.endpoint.samlmetadata,com.ibm.ws.security.saml.sso_fat.jaxrs,com.ibm.ws.security.saml.sso_fat.jaxrs.config,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.spnego_fat,com.ibm.ws.security.thread.identity_fat,com.ibm.ws.security.thread.zos.context_zfat,com.ibm.ws.security.thread.zos_zfat,com.ibm.ws.security.token.ltpa_fat,com.ibm.ws.security.utility_fat,com.ibm.ws.security.wim.scim_fat,com.ibm.ws.security.wim.scim_zfat,com.ibm.ws.security_fat,com.ibm.ws.security_fat.features,com.ibm.ws.webcontainer.security.feature_fat,com.ibm.ws.webcontainer.security.jacc.1.5_fat,com.ibm.ws.webcontainer.security_fat,com.ibm.ws.webcontainer.security_fat.features,com.ibm.ws.webcontainer.security_fat.passwordutil,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.sipcontainer_fat

**Comment by utle** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.spnego_fat,com.ibm.websphere.security_fat,com.ibm.ws.collective.security_fat,com.ibm.ws.collective.security_zfat,com.ibm.ws.ejbcontainer.security.3.2_fat,com.ibm.ws.ejbcontainer.security_fat,com.ibm.ws.ejbcontainer.security_fat.features,com.ibm.ws.security.audit_fat.config,com.ibm.ws.security.audit_fat.example,com.ibm.ws.security.audit_fat.features,com.ibm.ws.security.auth.data_fat,com.ibm.ws.security.authentication.builtin_fat,com.ibm.ws.security.authentication.tai_fat,com.ibm.ws.security.authorization.saf_zfat,com.ibm.ws.security.client_fat_java7,com.ibm.ws.security.context_fat,com.ibm.ws.security.credentials.saf_zfat,com.ibm.ws.security.csiv2_fat,com.ibm.ws.security.java2sec_fat,com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.oauth_fat,com.ibm.ws.security.oidc.client_fat,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.quickstart_fat,com.ibm.ws.security.registry.saf_zfat,com.ibm.ws.security.registry_fat,com.ibm.ws.security.saml.sso_fat.config,com.ibm.ws.security.saml.sso_fat.config.mapToUserRegistry,com.ibm.ws.security.saml.sso_fat.endpoint.samlmetadata,com.ibm.ws.security.saml.sso_fat.jaxrs,com.ibm.ws.security.saml.sso_fat.jaxrs.config,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.spnego_fat,com.ibm.ws.security.thread.identity_fat,com.ibm.ws.security.thread.zos.context_zfat,com.ibm.ws.security.thread.zos_zfat,com.ibm.ws.security.token.ltpa_fat,com.ibm.ws.security.utility_fat,com.ibm.ws.security.wim.scim_fat,com.ibm.ws.security.wim.scim_zfat,com.ibm.ws.security_fat,com.ibm.ws.security_fat.features,com.ibm.ws.webcontainer.security.feature_fat,com.ibm.ws.webcontainer.security.jacc.1.5_fat,com.ibm.ws.webcontainer.security_fat,com.ibm.ws.webcontainer.security_fat.features,com.ibm.ws.webcontainer.security_fat.passwordutil,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.sipcontainer_fat

**Comment by utle** [comment]
The SIP TAI test case now passed but we see other SIP FAT tests failure with connection refused which I don't think my code change cause these failures. I have confirmed with @TiborBeres. I am going to request another full FAT PB for SIP.

com.ibm.jmx.sip.utils.Utils                                  W Exception occured while connection to the MBean server:
java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)

**Comment by utle** [comment]
The two tests failed in bucket com.ibm.ws.webcontainer.security.jacc15.fat.RunAsTest are the same with defect 
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=277364

**Comment by utle** [comment]
The other FAT failure should not cause my code change to give customer select when to initialize their customer TAI either at the server startup or at the first incoming request.

**Review by arkarkala — APPROVED** [review]

**Comment by utle** [comment]
run-bot-libby

**Comment by utle** [comment]
#run-libby-bot

**Pull request merged**